extensions/nodes/nodes.go

```diff
@@ -0,0 +1,92 @@
+package nodes
+
+import (
+	"errors"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/kubeapi/secrets"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	initNodeLabelKey      = "rke.cattle.io/init-node"
+	local                 = "local"
+	machinePlanSecretType = "rke.cattle.io/machine-plan"
+	machineNameSteveLabel = "rke.cattle.io/machine-name"
+)
+
+// GetInitMachine accepts a client and clusterID and returns the "init node" machine
+// object for rke2/k3s clusters
+func GetInitMachine(client *rancher.Client, clusterID string) (*v1.SteveAPIObject, error) {
+	logrus.Info("Retrieving list of secrets...")
+	secretList, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	machinePlans := []corev1.Secret{}
+
+	logrus.Info("Identifying machine-plan secrets...")
+	for _, secret := range secretList.Items {
+		if secret.Type == machinePlanSecretType {
+			logrus.Infof("Found machine-plan secret: %s", secret.Name)
+			machinePlans = append(machinePlans, secret)
+		}
+	}
+
+	initNodeMachineName := ""
+
+	logrus.Info("Identifying init node machine plan...")
+	for _, machinePlan := range machinePlans {
+		v, ok := machinePlan.ObjectMeta.Labels[initNodeLabelKey]
+		if ok && v == "true" {
+			logrus.Infof("Found init node machine plan: %s", machinePlan.Name)
+			initNodeMachineName = machinePlan.Labels[machineNameSteveLabel]
+		}
+	}
```

Comment:
why not move what's in this for loop to line 36 instead? you could also break in the if statement since you're done looking for initNode.